apis/status/v1beta1/constraintpodstatus_types.go

```diff
@@ -97,11 +97,11 @@ func NewConstraintStatusForPod(pod *corev1.Pod, podOwnershipMode PodOwnershipMod
 		// the template name is the lower-case of the constraint kind
 		ConstraintTemplateNameLabel: strings.ToLower(constraint.GetKind()),
 	})
-	if podOwnershipMode == PodOwnershipEnabled {
-		if err := controllerutil.SetOwnerReference(pod, obj, scheme); err != nil {
-			return nil, err
-		}
+
+	if err = controllerutil.SetOwnerReference(pod, obj, scheme); err != nil {
```

Comment:
I missed the conversation that motivated this PR so don't have full context.

When working on GK, I often run the code locally on my machine pointed at a remote cluster. If I read correctly, this will only be possible by setting `POD_NAME` and `POD_NAMESPACE` to point to a valid pod on the cluster?

Reply:
No. The original code sort of obfuscated how this works.

1. To function properly, `SetOwnerReference` requires its first argument (`pod` in this case) to have `metadata.name` and `metadata.uid` set. Otherwise future API Server calls with `obj` fail with confusing errors.
2. If `--debug-use-fake-pod` is enabled, then we unconditionally set the `POD_NAME` env variable to a valid value before instantiating the fake Pod. So Pod is guaranteed to be a valid object.
3. `util.GetPodNamespace` falls back to `gatekeeper-system` if the env variable is not set. This prevents various crashes which can happen if the controller's Pod does not have `metadata.namespace` set.
4. Finally, the new `fakes.Pod` method sets a dummy `metadata.uid` which satisfies the fake API Server used in testing. The fake API Server validates that requests have `metadata.uid`, but does not ensure the UID is meaningful or valid.

Thus, all information required for the Pod to be used in debug mode is set.